lib/Value/AnswerChecker.pm

```diff
@@ -1721,6 +1721,7 @@ sub cmp_postfilter {
   $ans->{_filter_name} = "produce_equivalence_message";
   return $ans if $ans->{ans_message}; # don't overwrite other messages
   return $ans unless defined($ans->{prev_ans}); # if prefilters are erased, don't do this check
+  return $ans if ( defined( $ans->{bypass_equivalence_test} ) && $ans->{bypass_equivalence_test} );
```

Comment:
It is simpler here if you just use:
```perl
return $ans if ( $ans->{bypass_equivalence_test} );
```
If `$ans->{bypass_equivalence_test}` is not defined or is defined and is a false value, then the conditional will be false as desired here.
Perl doesn't emit an error or even a warning if the hash key is not defined in this situation.
The same is true where you check this condition in the file PGfunctionevaluators.pl.

I also don't like the extra spaces in the parentheses.  Of course this is just my coding style preference.  I generally am of the opinion that programming code should follow standard grammatical syntax rules to the extent that the programming language allows.

Reply:
Thanks - cleaned up (with a forced push) to use the simpler test. Sorry, but I am too used to worrying about undefined values... but as you said and testing confirmed -  no warning is issued without the test if the hash has this key.

Reply:
Looks good.

You even removed the spaces around the parenthetical.  There are many that prefer that coding style.